### PR TITLE
plan, executor: replace sort with logicalSort and physicalSort

### PR DESCRIPTION
**Pull request by hanfei1991**
No longer reuse sort for physical plan converting. We use two plans :  one is PhysicalSort which represents  memory sort, and another is NominalSort whose sort property is guaranteed by its child. Phantom sort is a fake operator which will not exist in final physical operator tree.

### CODE REVIEW

**Comment by coocood** [comment]
Is the name `PhantomSort` original?

**Comment by hanfei1991** [comment]
@coocood no, that's a whim. But I think it's suitable.

**Comment by coocood** [comment]
How about `NominalSort`.

**Comment by hanfei1991** [comment]
@coocood Good name !

File: plan/gen_physical_plans.go [hunk 1]
```diff
@@ -412,3 +412,29 @@ func (p *LogicalSelection) generatePhysicalPlans() []PhysicalPlan {
...
 }
+
+func (p *LogicalSort) getPhysicalSort() *PhysicalSort {
+	ps := PhysicalSort{ByItems: p.ByItems}.init(p.ctx)
+	ps.profile = p.profile
+	ps.SetSchema(p.schema)
+	return ps
+}
+
+func (p *LogicalSort) getPhantomSort() *NominalSort {
```
Comment (coocood) [inline plan/gen_physical_plans.go]: Update the name.

**Comment by coocood** [comment]
LGTM

**Pull request merged**